TodoListWebApp/Controllers/HomeController.cs

```diff
@@ -36,81 +36,84 @@ public ActionResult Index()
+                    HttpResponseMessage response = await client.SendAsync(request);
+
+                    //
+                    // Return the To Do List in the view.
+                    //
+                    if (response.IsSuccessStatusCode)
+                    {
+                        List<Dictionary<String, String>> responseElements = new List<Dictionary<String, String>>();
+                        JsonSerializerSettings settings = new JsonSerializerSettings();
+                        String responseString = await response.Content.ReadAsStringAsync();
+                        // [jelled] Untested but something along these lines should be better:
+                        atClaims = JsonConvert.DeserializeObject<List<ClaimInfo>>(responseString, settings);
+                        //responseElements = JsonConvert.DeserializeObject<List<Dictionary<String, String>>>(responseString, settings);
+
+                        //foreach (var claim in responseElements)
+                        //{
+                        //    ClaimInfo ci = new ClaimInfo();
+                        //    ci.ClaimType = claim["ClaimType"];
+                        //    ci.Value = claim["Value"];
+                        //    ci.ValueType = claim["ValueType"].Substring(claim["ValueType"].IndexOf('#') + 1);
+                        //    ci.SubjectName = ((claim["SubjectName"]) != null && (String.IsNullOrEmpty(claim["SubjectName"])) == false) ? claim["SubjectName"] : "Null";
+                        //    ci.IssuerName = !String.IsNullOrEmpty(claim["IssuerName"]) ? claim["IssuerName"] : "Null";
+
+                        //    atClaims.Add(ci);
+                        //}
 
-                    foreach (var claim in responseElements)
+                    }
+                    else
                     {
-                        ClaimInfo ci = new ClaimInfo();
-                        ci.ClaimType = claim["ClaimType"];
-                        ci.Value = claim["Value"];
-                        ci.ValueType = claim["ValueType"].Substring(claim["ValueType"].IndexOf('#') + 1);
-                        ci.SubjectName = ((claim["SubjectName"]) != null && (String.IsNullOrEmpty(claim["SubjectName"])) == false) ? claim["SubjectName"] : "Null";
-                        ci.IssuerName = !String.IsNullOrEmpty(claim["IssuerName"]) ? claim["IssuerName"] : "Null";
-
-                        ATresults.Add(ci);
+                        // [jelled] Show the error somehow.
                     }
-
-                    ViewBag.ATclaims = ATresults;
                 }
-                else
+                catch (AdalSilentTokenAcquisitionException)
                 {
-                    ViewBag.ATclaims = new List<ClaimInfo>();
+                    // [jelled] In case the exception is because acquiring the token silently failed, then the cached token is probably invalid
```

Comment:
I fully agree. And for the actual TodoList page this is the case. But from a demonstration perspective I want this page to only show the claims the user has. If he doesn't have any, I'm not showing those. Theoretically spoken, I could have an ID Token here, but not an Access Token. Then I want this page to reflect that.